### ISSUE TITLE
Add failover in the 

### ISSUE THREAD

**Issue by julienfr112** [ClickHouse/clickhouse-rs#43]
Clickhouse can ensure high availibility with replicated tables between servers
From the client point of view, it may be usefull to be able to rotate between multiple hosts.

For example, most postgres client support failover, and it's very usefull.

**Comment by loyd** [comment]
Good idea. I think we need to provide a more flexible interface to specify not only failovers but also shards. But even an initial simple implementation by providing multiple IP/hosts is a good start.
